tests/unit/controllers/admin/seeder-test.js

```diff
@@ -1,119 +1,113 @@
-  });
+  test('generateLibraries action', async function(assert) {
+    assert.expect(4);
+
+    const controller = this.owner.lookup('controller:admin/seeder');
+    controller._saveRandomLibrary = stub().resolves();
 
-  test('generateLibraries action', function(assert) {
-    const done = assert.async();
-    const { controller } = this;
     controller.send('generateLibraries', 10);
-    assert.expect(4);
-    assert.equal(controller.get('generateLibrariesInProgress'), true);
-    assert.equal(controller._saveRandomLibrary.callCount, 10);
 
-    next(() => {
-      assert.equal(controller.get('generateLibrariesInProgress'), false);
-      assert.equal(controller.get('libDone'), true);
-      done();
-    });
+    await assert.equal(controller.generateLibrariesInProgress, true);
+    await assert.equal(controller._saveRandomLibrary.callCount, 10);
+    await assert.equal(controller.generateLibrariesInProgress, false);
+    await assert.equal(controller.libDone, true);
   });
 
-  test('deleteLibraries action', function(assert) {
-    const done = assert.async();
-    const { controller } = this;
-    controller.send('deleteLibraries');
+  test('deleteLibraries action', async function(assert) {
     assert.expect(4);
-    assert.equal(controller.get('deleteLibrariesInProgress'), true);
-    assert.ok(controller._destroyAll.calledOnceWith('libraries'));
 
-    next(() => {
-      assert.equal(controller.get('libDelDone'), true);
-      assert.equal(controller.get('deleteLibrariesInProgress'), false);
-      done();
-    });
+    const controller = this.owner.lookup('controller:admin/seeder');
+    controller._destroyAll = stub().resolves();
+    controller.model = { libraries: 'libraryRecords' };
+
+    controller.send('deleteLibraries');
+
+    await assert.equal(controller.deleteLibrariesInProgress, true);
+    await assert.ok(controller._destroyAll.calledOnceWith('libraryRecords'));
+    await assert.equal(controller.libDelDone, true);
+    await assert.equal(controller.deleteLibrariesInProgress, false);
   });
 
-  test('generateBooksAndAuthors action', function(assert) {
-    const done = assert.async();
-    const { controller } = this;
-    controller.send('generateBooksAndAuthors', 10);
+  test('generateBooksAndAuthors action', async function(assert) {
     assert.expect(5);
-    assert.equal(controller.get('generateBooksInProgress'), true);
-    assert.equal(controller._saveRandomAuthor.callCount, 10);
-
-    next(() => {
-      assert.ok(controller._generateSomeBooks.calledWith('newAuthor'));
-      assert.equal(controller.get('authDone'), true);
-      assert.equal(controller.get('generateBooksInProgress'), false);
-      done();
+
+    const controller = this.owner.lookup('controller:admin/seeder');
+    controller._saveRandomAuthor = stub().resolves('newAuthor');
+    controller._generateSomeBooks = stub().resolves();
+
+    controller.send('generateBooksAndAuthors', 10);
```

Comment:
```js
controller.generateBooksAndAuthors(10);
```

Reply:
the official guide uses .send, in their examples... is there any difference? what do you think?

Reply:
The official guide indeed uses `send` for [testing controller actions](https://guides.emberjs.com/release/testing/testing-controllers/#toc_testing-controller-actions).

My understanding was flawed, it turns out that the `ActionHandler` [is available](https://github.com/emberjs/ember.js/blob/v3.17.0/packages/%40ember/-internals/runtime/lib/mixins/action_handler.js#L9-L10) on `Route`, `Component`, _and_ `Controller` and it's been this way for years. Thanks for the enlightenment!